opm/autodiff/StandardWell_impl.hpp

```diff
@@ -22,6 +22,28 @@
 
 namespace Opm
 {
+namespace Detail
+{
+    //! calculates ret = A^T * B
+    template< class K, int m, int n, int p >
+    static inline void multMatrixTransposed ( const Dune::FieldMatrix< K, n, m > &A,
```

Comment:
This crucial function should have a unit test. It looks correct to me, but it is something that is very testable with a few random matrices.

(Also, I prefer putting the ampersands next to the type, not the variable, as is most common in C++.)